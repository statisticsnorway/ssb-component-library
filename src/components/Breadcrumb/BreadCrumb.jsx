import React from 'react';
import PropTypes from 'prop-types';
import Link from '../Link';

const Breadcrumb = ({ className, items }) => (
	<div className={`ssb-breadcrumbs${className ? ` ${className}` : ''}`}>
		{items.map(item => (
			item.link ? (
				<div key={item.text}>
					<Link href={item.link}>{item.text}</Link>
					&nbsp;/&nbsp;
				</div>
			) : (
				<span key={item.text}>{item.text}</span>
			)
		))}
	</div>
);

Breadcrumb.propTypes = {
	className: PropTypes.string,
	items: PropTypes.arrayOf(PropTypes.shape({
		link: PropTypes.string,
		text: PropTypes.string.isRequired,
	})).isRequired,
};

export default Breadcrumb;
