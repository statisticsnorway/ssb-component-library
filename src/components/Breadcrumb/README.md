Breadcrumb
========

> Breadcrumbs are a navigational aid used to provide the user with the information of their location on the website
### Usage


#### HTML

```html
<div class="ssb-breadcrumbs">
    <div>
        <a class="ssb-link" href=" ">
            <span class="link-text">breadcrumb 1</span>
        </a>
        &nbsp;/&nbsp;
    </div>
    <div>
        <a class="ssb-link" href=" ">
            <span class="link-text">breadcrumb 2</span>
        </a>&nbsp;/&nbsp;
    </div>
    <span>breadcrumb 3</span>
</div>

```

#### React

```jsx harmony
const items = [
	{
		link: ' ',
		text: 'breadcrumb 1',
	},
	{
		link: ' ',
		text: 'breadcrumb 2',
	},
	{
		text: 'breadcrumb 3',
	},
];

  <Breadcrumb items={items} />
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| items   | Required array of objects | 	A list of objects. Object key text is required, link is optional|

__Dependencies__
 - [Link](../Link)
