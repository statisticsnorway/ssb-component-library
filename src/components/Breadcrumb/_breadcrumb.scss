.ssb-breadcrumbs {
  @include open-sans;
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  font-size: 16px;
  font-stretch: normal;
  line-height: 1.25;
}
