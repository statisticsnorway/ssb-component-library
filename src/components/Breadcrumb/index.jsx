import React from 'react';
import PropTypes from 'prop-types';
import Link from '../Link';

const Breadcrumb = ({ items }) => (
	<div className="ssb-breadcrumbs">
		{items.map(item => (
			item.link ? (
				<div key={item.text}>
					<Link href={item.link}>{item.text}</Link>
					&nbsp;/&nbsp;
				</div>
			) : (
				<span key={item.text}>{item.text}</span>
			)
		))}
	</div>
);

Breadcrumb.propTypes = {
	items: PropTypes.arrayOf(PropTypes.shape({
		link: PropTypes.string,
		text: PropTypes.string.isRequired,
	})).isRequired,
};

export default Breadcrumb;
