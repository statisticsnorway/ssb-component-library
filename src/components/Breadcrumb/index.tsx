import React from 'react'
import { ArrowLeft } from 'react-feather'
import Link from '../Link'
import classNames from '../../utils/utils'

interface Item {
  link: string
  text: string
}
interface BreadcrumbProps {
  className?: string
  items: Item[]
  mobileCompressedView?: boolean
}

const Breadcrumb: React.FC<BreadcrumbProps> = ({ className, items, mobileCompressedView = false }) => {
  const shouldShowCompressed = mobileCompressedView && items.length > 1

  return (
    <div className={classNames('ssb-breadcrumbs', className)}>
      {shouldShowCompressed ? (
        <Link href={items[items.length - 2].link} icon={<ArrowLeft size={20} />}>
          {items[items.length - 2].text}
        </Link>
      ) : (
        <>
          {items.slice(0, -1).map((item) => (
            <div key={item.text}>
              <Link href={item.link}>{item.text}</Link>
              &nbsp;/&nbsp;
            </div>
          ))}
          <span>{items[items.length - 1].text}</span>
        </>
      )}
    </div>
  )
}

export default Breadcrumb
