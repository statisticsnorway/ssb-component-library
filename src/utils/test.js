import fs from 'fs';
import path from 'path';
import { render as rtlRender } from '@testing-library/react';

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8');

export function render(ui, options) {
	const view = rtlRender(ui, options);

	const styleElement = document.createElement('style');
	styleElement.innerHTML = styles;
	document.body.appendChild(styleElement);

	return view;
}
