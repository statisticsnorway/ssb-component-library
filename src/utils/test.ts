import fs from 'fs'
import path from 'path'
import { RenderOptions, RenderResult, render as rtlRender } from '@testing-library/react'
import { ReactNode } from 'react'

export * from '@testing-library/react'

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8')

export function render(ui: ReactNode, options?: RenderOptions): RenderResult {
  const view = rtlRender(ui, options)

  const styleElement = document.createElement('style')
  styleElement.innerHTML = styles
  document.body.appendChild(styleElement)

  return view
}
