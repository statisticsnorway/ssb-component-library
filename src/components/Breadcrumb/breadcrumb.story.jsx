import React from 'react'
import Breadcrumb from './index'

const mockedItems = [
  {
    link: ' ',
    text: 'breadcrumb 1',
  },
  {
    link: ' ',
    text: 'breadcrumb 2',
  },
  {
    text: 'breadcrumb 3',
  },
]

export default {
  title: 'Breadcrumb',
}

export const Default = () => (
  <div>
    <Breadcrumb items={mockedItems} />
  </div>
)
