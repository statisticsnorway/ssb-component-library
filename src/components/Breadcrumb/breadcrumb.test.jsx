import React from 'react'
import { screen, render } from '../../utils/test'
import Breadcrumb from './index'

const items = [
  {
    link: ' ',
    text: 'breadcrumb 1',
  },
  {
    link: ' ',
    text: 'breadcrumb 2',
  },
  {
    text: 'breadcrumb 3',
  },
]

describe('Breadcrumb component', () => {
  test('Matches the snapshot', () => {
    const { asFragment } = render(<Breadcrumb items={items} />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('Renders all items', async () => {
    render(<Breadcrumb items={items} />)
    expect(await screen.findByText(items[0].text)).toBeVisible()
    expect(await screen.findByText(items[1].text)).toBeVisible()
    expect(await screen.findByText(items[2].text)).toBeVisible()
  })
})
