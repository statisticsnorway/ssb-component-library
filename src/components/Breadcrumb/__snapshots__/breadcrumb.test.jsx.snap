// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ssb-breadcrumbs"
  >
    <div>
      <a
        class="ssb-link"
        href=" "
      >
        <span
          class="link-text"
        >
          breadcrumb 1
        </span>
      </a>
       / 
    </div>
    <div>
      <a
        class="ssb-link"
        href=" "
      >
        <span
          class="link-text"
        >
          breadcrumb 2
        </span>
      </a>
       / 
    </div>
    <span>
      breadcrumb 3
    </span>
  </div>
</DocumentFragment>
`;
